gin.go

```diff
@@ -305,12 +314,35 @@ func iterate(path, method string, routes RoutesInfo, root *node) RoutesInfo {
 func (engine *Engine) Run(addr ...string) (err error) {
 	defer func() { debugPrintError(err) }()
 
+	trustedCIDRs, err := engine.prepareCIDR()
+	if err != nil {
+		return err
+	}
+	engine.trustedCIDRs = trustedCIDRs
 	address := resolveAddress(addr)
 	debugPrint("Listening and serving HTTP on %s\n", address)
 	err = http.ListenAndServe(address, engine)
 	return
 }
 
+func (engine *Engine) prepareCIDR() ([]*net.IPNet, error) {
+	if engine.TrustedProxies != nil {
+		cidr := make([]*net.IPNet, len(engine.TrustedProxies), 0)
+		for _, trustedProxy := range engine.TrustedProxies {
+			if strings.Contains(trustedProxy, "/") {
```

Comment:
Seems worth applying some regex magic here?

Reply:
Maybe there are more corner cases that we originally thought. 😅 

self-note: ipv6 compatibility (?)

Reply:
@austinheap solved at https://github.com/gin-gonic/gin/pull/2632/commits/7e649c347fed4f4322d5a3eb893b8eff17c6d843